### PR TITLE
Add api_management_gateway_hostname_configuration

### PR DESCRIPTION
**Pull request by dvob**
Add support for hostConfigurations in API management gateways.
For this I took more or less the `api_management_gateway` resource and data source as a template and adjusted it accordingly.
Unfortunately I can't run the integration tests because I don't have a client_id and client_secret to access an Azure tenant.

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @dvob - overall this looks good but we need to fix up the tests:

image

**Comment by manicminer** [comment]
Removed the resource as it was already added in https://github.com/hashicorp/terraform-provider-azurerm/pull/17962 with slightly different name, and renamed data source to match

File: website/docs/d/api_management_gateway_host_name_configuration.html.markdown [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported: 
+
+* `id` - The ID of the API Management Gateway Host Configuration.
+
+* `certificate_id` - The ID of the certificate.
+
+* `hostname` - The hostname.
+
```
Comment (stephybun) [inline website/docs/d/api_management_gateway_host_name_configuration.html.markdown:53]: It looks like there are some attributes missing e.g. `request_client_certificate_enabled`, `http2_enabled`, `tls10_enabled` and `tls11_enabled`.

File: website/docs/d/api_management_gateway_host_name_configuration.html.markdown [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+## Attributes Reference
+
+In addition to the Arguments listed above - the following Attributes are exported: 
+
+* `id` - The ID of the API Management Gateway Host Configuration.
+
+* `certificate_id` - The ID of the certificate.
+
+* `hostname` - The hostname.
```
[the remark this answers is not in the document]
Comment (manicminer) [reply website/docs/d/api_management_gateway_host_name_configuration.html.markdown]: Good catch thanks, have updated the docs

File: website/docs/d/api_management_gateway_host_name_configuration.html.markdown (hunk 1 above)
Comment (manicminer) [reply website/docs/d/api_management_gateway_host_name_configuration.html.markdown:64]: Docs updated 👍

**Review by stephybun — APPROVED** [review]
Thanks @dvob and @manicminer LGTM 🍕

**Comment by dvob** [comment]
@manicminer Thank you for finalizing this PR

**Pull request merged**